tests/js/chartAuthoringTests.js

```diff
@@ -155,8 +159,50 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
         jqUnit.assertDeepEq("The reset has an aria-controls attribute properly referencing the form, pie, legend and total", resetAriaControlsAttr, dataEntryFormlId + " " + legendTableId + " " + pieChartPieId + " " + dataEntryFormTotalId);
     };
 
+    // Verify that the updateDataEntryPanel function updates the UI
+    floe.tests.chartAuthoringTester.verifyUpdateDataEntryPanel = function (that) {
+        that.updateDataEntryPanelFromDataSet(floe.tests.chartAuthoring.updateDataSet);
+
+        var dataEntries = that.chartAuthoringInterface.dataEntryPanel.locate("dataEntry");
+
+        var dataEntryLabelSelector = that.chartAuthoringInterface.dataEntryPanel.dataEntry.options.selectors.label;
+        var dataEntryValueSelector = that.chartAuthoringInterface.dataEntryPanel.dataEntry.options.selectors.value;
+
+        dataEntries.each(function (idx) {
+
+            // Confirm that each entry in updateDataSet has a matching dataEntry
+            // in the panel UI, or that the "matching" dataEntry is blank when
+            // the dataset is shorter than the number of possible data entries
+
+            var expectedData = idx < floe.tests.chartAuthoring.updateDataSet.length ? floe.tests.chartAuthoring.updateDataSet[idx] : expectedData = {label: "", value: ""};
+
+            jqUnit.assertEquals("Testing updated dataEntryPanel UI - displayed label at position " + idx + " matches updated data set label", $(this).find(dataEntryLabelSelector).val(), String(expectedData.label));
```

Comment:
While my memory is a bit hazy on the exact conversation, I think we discussed a similar issue when implementing the original `updateDataEntryPanelFromDataSet` functionality (which this is testing) back in November. The way the test looping of the `dataEntry` elements works is modelled on how the function being tested works in terms of getting all the individual `dataEntry` components from the `dataEntryPanel` and performing some action on each of them. I think [this is the final commit from November](https://github.com/waharnum/chartAuthoring/commit/6ad54c317c5c6db2c9821b8a226bf3c59f4222c3) after we looked at the issue closely and concluded that using `$(this).find)` was the best option, given Infusion doesn't have a method at the moment to easily loop over a set of `dynamicComponents`.

I think this is also already scoped to `that.container` because the original set of jQuery elements being looped by `dataEntries.each` is found by `that.chartAuthoringInterface.dataEntryPanel.locate("dataEntry");`